modules/tinymce/src/core/main/ts/delete/TableDeleteUtils.ts

```diff
@@ -20,24 +20,45 @@ const isRootFromElement = (root: SugarElement<Node>): IsRootFn =>
 const getTableCells = (table: SugarElement<HTMLTableElement>): SugarElement<HTMLTableCellElement>[] =>
   SelectorFilter.descendants<HTMLTableCellElement>(table, 'td,th');
 
+const getTable = (node: Node, isRoot: IsRootFn) => TableCellSelection.getClosestTable(SugarElement.fromDom(node), isRoot);
+
+const selectionInTableWithNestedTable = (details: TableSelectionDetails, rng: Range, isRoot: IsRootFn): TableSelectionDetails => {
+  return Optionals.lift3(details.startTable, details.endTable, getTable(rng.commonAncestorContainer, isRoot), (startTable, endTable, ancestorTable) => {
```

Comment:
Could this be simplified by simply checking if startTable contains endTable or endTable constains startTable or is that just more confusing?

Reply:
@spocke I changed it but it still seems a very confusing thing (anyway is better that my previous implementation :sweat_smile: ), if I find a better approach I'll change it otherwise I'll merge with this version

Reply:
ok, I removed the check on the same table since is implicit using `PredicateExists.descendant` and also moved the other way around the condition in the return to don't use the negation, I think I'll be not able to improve anymore, I'll merge it as soon as the CI finish

Reply:
`PredicateExists.descendant` is quite slow. `Compare.contains` is what Spocke was talking about. I wish I'd seen this earlier 😞